File: CSharp/DP.Domain/Samples/Interpreter/PayExpression.cs
using System;

namespace DP.Domain.Samples.Interpreter
{
    /// <summary>
    /// Expression
    /// </summary>
    public class PayExpression : IExpression
    {
        public void Interpret(Context context)
        {
            context.Output.Customer = context.Input.Substring(38, 20).Trim();
            context.Output.PayAmout = Decimal.Parse(context.Input.Substring(58, 8).Trim());
            context.Output.PayOn = DateTime.Parse(context.Input.Substring(66, 19));
        }
    }
}

File: Python/Samples/Interpreter/PayExpression.py
from datetime import datetime
from decimal import Decimal 
from Interpreter_models import PayData, Vip, Store, Context
from Expression import Expression

class PayExpression(Expression):
     def interpret(self,context=Context):
         context.output.customer = context.input[38:50].strip()
         context.output.payAmt = Decimal(context.input[58:64].strip())
         context.output.payOn = datetime.strptime(context.input[66:85],"%Y-%m-%d %H:%M:%S")